utils/src/atomics/atomic_traits.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Atomic traits. Unlike traits from the `atomic_traits` crate, these are implemented
+//! for `loom` types too.
+//!
+//! Note: functions that are not implemented by `loom` atomics (such as `get_mut`) are missing
+//! here too.
+//! We also omit deprecated functions, such as `compare_and_swap`, and don't support `AtomicPtr`.
+
+use crate::concurrency_impl::sync::atomic::{
+    AtomicBool, AtomicI16, AtomicI32, AtomicI64, AtomicI8, AtomicIsize, AtomicU16, AtomicU32,
+    AtomicU64, AtomicU8, AtomicUsize, Ordering,
+};
+
+/// A trait that contains basic atomic operations common for all atomic types.
+pub trait Atomic {
```

Comment:
I'm a little surprised we're not using `atomic_traits::Atomic`... it's already in utils.

Reply:
Initially I was using them, but then I found out that they are not implemented for `loom`'s atomic types. So the alternative would be to abandon the idea of using `loom`'s atomics.